tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/RightToWorkValidator.java

```diff
@@ -83,14 +89,37 @@ private void checkSettled(RightToWorkDTO dto, List<FieldError> fieldErrors) {
     }
   }
 
-  private void checkVisaDates(RightToWorkDTO dto, List<FieldError> fieldErrors) {
-    LocalDate visaIssued = dto.getVisaIssued();
-    LocalDate visaValidTo = dto.getVisaValidTo();
+  private void checkVisaDates(List<FieldError> fieldErrors, PersonDTO personDto) {
 
-    if (visaIssued != null && visaValidTo != null && visaIssued.isAfter(visaValidTo)) {
-      FieldError fieldError =
-          new FieldError(DTO_NAME, "visaIssued", "visaIssued must be before visaValidTo.");
-      fieldErrors.add(fieldError);
+    if (personDto != null) {
+      String field = "visaIssued";
+      LocalDate visaIssued = personDto.getRightToWork().getVisaIssued();
+      LocalDate visaValidTo = personDto.getRightToWork().getVisaValidTo();
```

Comment:
Nice work validating the person.  There is at least a theoretical chance of a NPE for `.getVisaIssued()`.  You could change the if clause to `personDto != null && personDto.getRightToWork() != null` or make this a "precondition" style block:
```java
  if (personDto == null || personDto.getRightToWork() != null) {
    return;
  }
```